R/Viewer.R

```diff
@@ -346,12 +347,12 @@ odbcConnectionActions.default <- function(connection) {
               tableName <- dbQuoteIdentifier(connection, firstTable$table_name)
 
               # add schema
-              if (!is.null(firstTable$table_schema) && nchar(firstTable$table_schema) > 0) {
+              if (!is.null(firstTable$table_schema) && !is.na(firstTable$table_schema) && nchar(firstTable$table_schema) > 0) {
```

Comment:
This level of type flexibility always makes me a bit nervous. Could we add some type checks higher up so we only need to check for one empty sentinel?

Reply:
Thanks Hadley:

As best as I can tell, the `is.null(...)` that was there to begin-with is not guarding anything; I think only the `is.na` is actually effective.

Kind of hesitant to remove the `is.null` since I am not sure why it was there in the first place.  But at the same time this is not in any critical code-path; we can revisit if remove it breaks something.